frontend/src/components/GroupFilterCard.tsx

```diff
@@ -0,0 +1,172 @@
+import {useTranslation} from 'react-i18next';
+import {NumberFormatter} from '../util/hooks';
+import {ScrollSyncPane} from 'react-scroll-sync';
+import List from '@mui/material/List';
+import ListItem from '@mui/material/ListItem';
+import ListItemText from '@mui/material/ListItemText';
+import {GroupFilter} from '../types/group';
+
+/**
+ * This is responsible for displaying an active group filter.
+ */
+export function GroupFilterCard(props: GroupFilterCardProps): JSX.Element | null {
+  const theme = useTheme();
+
+  return (
+    <Box
+      id={`scenario-card-${props.scenarioId}-group-filter-root-${props.groupFilterIndex}`}
+      key={props.groupFilter.name}
+      sx={{
+        padding: theme.spacing(2),
+        paddingLeft: `calc(${theme.spacing(2)} + 6px)`,
+        margin: '6px',
+        marginLeft: '0',
+        alignContent: 'center',
+        borderLeft: props.groupFilterIndex == 0 ? null : `1px solid`,
+        borderColor: props.groupFilterIndex == 0 ? null : 'divider',
+      }}
+    >
+      <Box
+        sx={{
+          display: 'flex',
+          alignItems: 'flex-end',
+          height: '3rem',
+          marginBottom: theme.spacing(1),
+        }}
+      >
+        <Typography
+          variant='h2'
+          sx={{
+            height: 'min-content',
+            fontWeight: 'bold',
+            fontSize: '13pt',
+          }}
+        >
+          {props.groupFilter.name}
+        </Typography>
+      </Box>
+      <GroupFilterCardCompartmentValues
+        groupFilter={props.groupFilter}
+        groupFilterIndex={props.groupFilterIndex}
+        scenarioId={props.scenarioId}
+      />
+    </Box>
+  );
+}
+
+/**
+ * This component renders all compartment values of a group filter in one column.
+ */
+function GroupFilterCardCompartmentValues(props: GroupFilterCardProps): JSX.Element | null {
+  const {t, i18n} = useTranslation();
+  const theme = useTheme();
+
+  const {formatNumber} = NumberFormatter(i18n.language, 1, 0);
+
+  const groupFilterList = useAppSelector((state) => state.dataSelection.groupFilters);
+  const day = useAppSelector((state) => state.dataSelection.date);
+  const node = useAppSelector((state) => state.dataSelection.district?.ags);
+  const selectedCompartment = useAppSelector((state) => state.dataSelection.compartment);
+  const compartmentsExpanded = useAppSelector((state) => state.dataSelection.compartmentsExpanded);
+  const compartments = useAppSelector((state) => state.scenarioList.compartments);
+
+  const {data: groupFilterData} = useGetSingleGroupFilterDataQuery(
+    {
+      id: props.scenarioId,
+      node: node,
+      day: day ?? '',
+      groupFilter: props.groupFilter,
+    },
+    {skip: !day}
+  );
+
+  const groupCompartmentsRef = useRef<Array<HTMLUListElement | null>>([]);
+
+  useEffect(() => {
+    if (groupFilterList) {
+      groupCompartmentsRef.current.slice(0, Object.values(groupFilterList).length);
+    }
+  }, [groupFilterList]);
```

Comment:
Add comments explaining why you do this or how the ref is used in the List component.